service/k8sclient/internal/kubectl/kubectl.go

```diff
@@ -23,33 +23,56 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"math/rand"
+	"net"
 	"os"
 	"os/exec"
 	"strconv"
 	"strings"
+	"sync"
+	"time"
 
+	"github.com/avast/retry-go"
+	gocache "github.com/patrickmn/go-cache"
 	"github.com/percona/pmm/utils/pdeathsig"
 	"github.com/pkg/errors"
 	"golang.org/x/sys/unix"
 
 	"github.com/percona-platform/dbaas-controller/utils/logger"
 )
 
+// Option represents kubectl option, it's meant to be set of bitmasks.
+type Option uint64
+
 const (
 	dbaasToolPath           = "/opt/dbaas-tools/bin"
 	defaultPmmServerKubectl = dbaasToolPath + "/kubectl-1.16"
 	defaultDevEnvKubectl    = "minikube kubectl --"
+	expirationTime          = time.Second * 10
+	// UseCacheOption is an option that turns on cache of kubectl commands.
+	UseCacheOption Option = 1
+	maxRetries            = 10
 )
 
+type proxyProcess struct {
```

Comment:
Can we extract everything related to proxy to a separate file?

Reply:
done